packages/core/src/xml-transformations/xslt-model-adapter.ts

```diff
@@ -0,0 +1,319 @@
+import {
+  StructureModelClass,
+  StructureModelPrimitiveType,
+  StructureModelProperty,
+  StructureModel,
+  StructureModelType,
+  StructureModelComplexType,
+} from "../structure-model";
+import {
+  XmlTransformation,
+  XmlTemplate,
+  XmlRootTemplate,
+  XmlMatch,
+  QName,
+  XmlClassMatch,
+  XmlLiteralMatch,
+  XmlTransformationInclude,
+} from "./xslt-model";
+
+import {
+  DataSpecification,
+  DataSpecificationArtefact,
+  DataSpecificationSchema,
+} from "../data-specification/model";
+
+import { XSD, OFN } from "../well-known";
+import { XSLT_LIFTING, XSLT_LOWERING } from "./xslt-vocabulary";
+
+export function structureModelToXslt(
+  specifications: { [iri: string]: DataSpecification },
+  specification: DataSpecification,
+  model: StructureModel
+): XmlTransformation {
+  const adapter = new XsltAdapter(specifications, specification, model);
+  return adapter.fromRoots(model.roots);
+}
+
+const anyUriType: StructureModelPrimitiveType = (function () {
+  const type = new StructureModelPrimitiveType();
+  type.dataType = XSD.anyURI;
+  return type;
+})();
+
+/**
+ * Map from common datatype URIs to XSD datatypes.
+ */
+const simpleTypeMap: Record<string, string> = {
+  [OFN.boolean]: XSD.boolean,
+  [OFN.date]: XSD.date,
+  [OFN.time]: XSD.time,
+  [OFN.dateTime]: XSD.dateTimeStamp,
+  [OFN.integer]: XSD.integer,
+  [OFN.decimal]: XSD.decimal,
+  [OFN.url]: XSD.anyURI,
+  [OFN.string]: XSD.string,
+};
```

Comment:
You have similar fragment of code in xml-schema-model-adapter:
```
const simpleTypeMap: Record<string, QName> = {
  [OFN.boolean]: ["xs", "boolean"],
  [OFN.date]: ["xs", "date"],
  [OFN.time]: ["xs", "time"],
  [OFN.dateTime]: ["xs", "dateTimeStamp"],
  [OFN.integer]: ["xs", "integer"],
  [OFN.decimal]: ["xs", "decimal"],
  [OFN.url]: ["xs", "anyURI"],
  [OFN.string]: ["xs", "string"],
  [OFN.text]: langStringName,
};````
Are the related? Is so they should be extracted to common place.

Reply:
Yeah, this and couple other pieces will be extracted to a common module.